src/Config/Variant.php

```diff
@@ -92,6 +102,23 @@ public function __toString(): string
         return $this->name;
     }
 
+    public function isAvailable(): bool
+    {
+        return $this->available;
+    }
+
+    public function setAvailable(bool $available): void
+    {
+        $this->available = $available;
+    }
+
+    public function isCreatable(): bool
+    {
+        return $this->width !== null
+            && $this->height !== null
+            && in_array($this->fit, self::AVAILABLE_FITS);
+    }
+
```

Comment:
This code does not belong here. It's more or less coupled to some concrete generator / flow

Reply:
I agree, it seems I have misunderstood where the coupling should appear. I'll move it instead to an event subscription. Question is how much we care about reporting that certain variants might not be available.

But all that still depends on whether we want variants to be creatable in general (see [comment](https://github.com/Setono/SyliusImagePlugin/pull/16#discussion_r818405141)).

Reply:
Imo they should be created (as they were before this PR), but done with a listener (as it was before). What other things to consider is up to you, but I think it should be possible to automatically create those variants in the Cloudflare interface